### DISCUSSION TITLE
I want to preview the pdf file on click

### DISCUSSION THREAD

**Discussion by mohandtharsha** [filamentphp/filament#9020] [Help]
I have resource called applicants in that i have added view resume link on click of modal is opening but on click of i want view the resume below is my code,<?php

namespace App\Filament\Employer\Resources;

use App\Filament\Employer\Resources\ApplicantsResource\Pages;
use App\Filament\Employer\Resources\ApplicantsResource\RelationManagers;
use App\Models\Applicants;
use App\Models\UserResumes;
use Filament\Forms;
use Illuminate\Support\Facades\Auth;
use Filament\Forms\Form;
use Filament\Resources\Resource;
use Filament\Tables;
use Filament\Tables\Table;
use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\SoftDeletingScope;
use Filament\Tables\Columns\TextColumn;
use Filament\Tables\Actions\ActionGroup;
use Filament\Tables\Actions\Action;
use Filament\Forms\Components\FileUpload;
use App\Http\Livewire\InterviewSetupComponent; // Import your Livewire component
use Filament\Tables\Actions\Modal;
use Filament\Forms\Components\TextInput;
use Filament\Forms\Components\Select;
use Filament\Forms\Components\Radio;
use Filament\Forms\Components\MarkdownEditor;
use Filament\Forms\Components\DatePicker;
use Filament\Forms\Components\DateTimePicker;
use Filament\Forms\Components\TimePicker;
use Filament\Forms\Components\Fieldset;
use App\Models\Interviews;
use Filament\Forms\Components\Hidden;
use Filament\Notifications\Notification;
use Illuminate\Support\Facades\Response;

class ApplicantsResource extends Resource
{
    protected static ?string $model = Applicants::class;

    protected static ?string $navigationIcon = 'heroicon-o-users';

    public static function form(Form $form): Form
    {
        return $form
            ->schema([

            ]);
    }

    public static function table(Table $table): Table
    {
        return $table
            ->columns([
                TextColumn::make('user.name')
                ->searchable()
                ->label('Applicant Name'),
                TextColumn::make('created_at')
                // ->searchable()
                ->label('Applied On:'),
                TextColumn::make('job.title')
                 ->label('Applied To:'),
                TextColumn::make('job.location')
                // ->searchable()
                ->label('Job Location:'),
                TextColumn::make('resume.resume_file')
                ->label('Resume:')
            ])
            ->filters([
                //
            ])
            ->actions([
                ActionGroup::make([
                    Tables\Actions\EditAction::make(),
                    Tables\Actions\DeleteAction::make(),
                    Tables\Actions\ViewAction::make(),
                    Action::make('Setup Interview')
                    ->form([
                        Hidden::make('id'), // Add a hidden field to store the applicant's ID
                        Select::make('duration')
                        ->required()
                        ->label('Duration')
                        ->options([
                            '15' => '15',
                            '30' => '30',
                            '45' => '45',
                            '60' => '60',
                            '90' => '90',
                            '120'=> '120',
                        ]),
                        Radio::make('format_type')
                        ->required()
                        ->label('Format')
                        ->options([
                            'video' => 'Video',
                            'phone' => 'Phone',
                        ])
                        ->live(),
                        TextInput::make('video_link')
                            ->label('Video Link')
                            ->required()
                            ->visible(fn ($get): bool => $get('format_type') === 'video'),
                        TextInput::make('phone')
                            ->label('Phone Number')
                            ->required()
                            ->visible(fn ($get): bool => $get('format_type') === 'phone'),
                        MarkdownEditor::make('message')
                            ->label('Message to applicant'),
                        Fieldset::make('Schedule a specific time')
                            ->schema([
                                DatePicker::make('interview_date')
                                ->label('Date'),
                                TimePicker::make('interview_time')
                                ->label('Time')
                            ])

                    ])
                    ->action(function (array $data,Applicants $record): void {
                        $user = Auth::user();
                        $applicants = Applicants::with('job')->get();
                        $result = Interviews::create([
                            'candidate_id' => $record->user_id,
                            'job_id' => $record->job_id,
                            'interviewer_id' => $user->id,
                            'interview_duration' => $data['duration'],
                            'interview_format' => $data['format_type'],
                            'video_link' => $data['video_link'] ?? null,
                            'phone_number' => $data['phone_number'] ?? null,
                            'interview_date' => $data['interview_date'],
                            'interview_time' => $data['interview_time'],
                            'message_to_applicant' => $data['message'],
                        ]);
                        if($result){
                            Notification::make()
                            ->title('Interview has been scheduled!')
                            ->success()
                            ->send();
                        }
                    })
                    ->slideOver(),
                    Action::make('viewResume')
                    ->form([
                        FileUpload::make('resume.resume_file')
                        ->multiple()
                        ->openable()
                    ])
                    ->action(function (array $data,Applicants $record): void {
                        $resume = UserResumes::where('id',$record->resume_id)->first();
                        // dd($resume->resume_file);
                        if ($resume) {
                            // Get the path to the resume file on your server
                            $resumePath = storage_path('app/public/resumes/' . $resume->resume_file);
                            // dd($resumePath);
                            // Check if the file exists
                            if (file_exists($resumePath)) {
                                // Use JavaScript to open the PDF in a new tab
                                $this->dispatchBrowserEvent('open-pdf', ['url' => asset('storage/resumes/' . $record->applicant->resume->resume_file)]);
                            }

                        }
                    })
                    // ->slideOver()
                    ->label('View Resume'),

                    // ->handler(Action\DownloadResume::class), // Use the DownloadResume action class

                    // Tables\Actions\SetupInterview::make(),
            ])
        ])
            ->bulkActions([
                Tables\Actions\BulkActionGroup::make([
                    Tables\Actions\DeleteBulkAction::make(),
                ]),
            ])
            ->emptyStateActions([
                // Tables\Actions\CreateAction::make(),
            ]);
    }

    public static function getRelations(): array
    {
        return [
            //
        ];
    }

    public static function getPages(): array
    {
        return [
            'index' => Pages\ListApplicants::route('/'),
            'create' => Pages\CreateApplicants::route('/create'),
            'edit' => Pages\EditApplicants::route('/{record}/edit'),
        ];
    }

    public function showInterviewSetupModal()
    {
        $this->openModal('interview-setup');
    }

    public function modalComponent()
    {
        return InterviewSetupComponent::class;
    }

    public function download($path,$file){
        return Response::download($path, $file);
    }
}

**Comment by leandrocfe** [comment]
You could [embed](https://stackoverflow.com/questions/291813/recommended-way-to-embed-pdf-in-html) the pdf on the modalContent

**Comment by photojo** [comment]
I use a uuid as well, so the pdf-preview could be actualized.
